Lib/test/test_msvcrt.py

```diff
@@ -64,35 +63,43 @@ def test_get_osfhandle(self):
 
 class TestConsoleIO(unittest.TestCase):
     def test_kbhit(self):
+        h = msvcrt.get_osfhandle(sys.stdin.fileno())
+        ctypes.windll.kernel32.FlushConsoleInputBuffer(h)
         self.assertEqual(msvcrt.kbhit(), 0)
 
     def test_getch(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getch(), b'c')
 
     def test_getwch(self):
-        stdin = open('CONIN$', 'r')
-        old_stdin = sys.stdin
-        try:
-            sys.stdin = stdin
-            write_input(stdin.buffer.raw, c_encoded)
-            self.assertEqual(msvcrt.getwch(), c)
-        finally:
-            sys.stdin = old_stdin
+        with open('CONIN$', 'rb', buffering=0) as stdin:
+            h = msvcrt.get_osfhandle(stdin.fileno())
+            ctypes.windll.kernel32.FlushConsoleInputBuffer(h)
```

Comment:
Maybe it should be exposed in msvcrt, maybe as a private function just for testing.